### PR TITLE
HBASE-26175 MetricsHBaseServer should record all kinds of Exceptions

### PR DESCRIPTION
**Pull request by sunhelly**

### CODE REVIEW

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 39s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 14s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   2m 15s |  master passed  |
| +1 :green_heart: |  compile  |   2m 29s |  master passed  |
| +1 :green_heart: |  checkstyle  |   0m 42s |  master passed  |
| +1 :green_heart: |  spotbugs  |   1m 33s |  master passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 10s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   2m  4s |  the patch passed  |
| +1 :green_heart: |  compile  |   2m 27s |  the patch passed  |
| +1 :green_heart: |  javac  |   2m 27s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   0m  7s |  The patch passed checkstyle in hbase-hadoop-compat  |
| +1 :green_heart: |  checkstyle  |   0m 33s |  hbase-server: The patch generated 0 new + 12 unchanged - 1 fixed = 12 total (was 13)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| -1 :x: |  hadoopcheck  |   6m 54s |  The patch causes 10 errors with Hadoop v3.2.2.  |
| +1 :green_heart: |  spotbugs  |   1m 43s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 14s |  The patch does not generate ASF License warnings.  |
|  |   |  27m 32s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4248/1/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/4248 |
| Optional Tests | dupname asflicense javac spotbugs hadoopcheck hbaseanti checkstyle compile |
| uname | Linux d8c0b5c9ed2a 5.4.0-1025-aws #25~18.04.1-Ubuntu SMP Fri Sep 11 12:03:04 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 53a7ca794a |
| Default Java | AdoptOpenJDK-1.8.0_282-b08 |
| hadoopcheck | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4248/1/artifact/yetus-general-check/output/patch-javac-3.2.2.txt |
| Max. process+thread count | 60 (vs. ulimit of 30000) |
| modules | C: hbase-hadoop-compat hbase-server U: . |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4248/1/console |
| versions | git=2.17.1 maven=3.6.3 spotbugs=4.2.2 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 58s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 15s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   2m 51s |  master passed  |
| +1 :green_heart: |  compile  |   1m  0s |  master passed  |
| +1 :green_heart: |  shadedjars  |   3m 42s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 41s |  master passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 13s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   2m 33s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m  0s |  the patch passed  |
| +1 :green_heart: |  javac  |   1m  0s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   3m 39s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 39s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   0m 38s |  hbase-hadoop-compat in the patch passed.  |
| +1 :green_heart: |  unit  | 177m 26s |  hbase-server in the patch passed.  |
|  |   | 197m 18s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4248/1/artifact/yetus-jdk11-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/4248 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux 3712b5c0fb7c 5.4.0-90-generic #101-Ubuntu SMP Fri Oct 15 20:00:55 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 53a7ca794a |
| Default Java | AdoptOpenJDK-11.0.10+9 |
|  Test Results | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4248/1/testReport/ |
| Max. process+thread count | 2497 (vs. ulimit of 30000) |
| modules | C: hbase-hadoop-compat hbase-server U: . |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4248/1/console |
| versions | git=2.17.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m  4s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 17s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   3m 39s |  master passed  |
| +1 :green_heart: |  compile  |   1m 19s |  master passed  |
| +1 :green_heart: |  shadedjars  |   5m 32s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 58s |  master passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 14s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   3m 24s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m 19s |  the patch passed  |
| +1 :green_heart: |  javac  |   1m 19s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   5m 24s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 51s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   0m 54s |  hbase-hadoop-compat in the patch passed.  |
| +1 :green_heart: |  unit  | 299m 17s |  hbase-server in the patch passed.  |
|  |   | 326m 21s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4248/1/artifact/yetus-jdk8-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/4248 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux 3ba32536c91c 5.4.0-90-generic #101-Ubuntu SMP Fri Oct 15 20:00:55 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 53a7ca794a |
| Default Java | AdoptOpenJDK-1.8.0_282-b08 |
|  Test Results | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4248/1/testReport/ |
| Max. process+thread count | 2265 (vs. ulimit of 30000) |
| modules | C: hbase-hadoop-compat hbase-server U: . |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4248/1/console |
| versions | git=2.17.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

File: hbase-server/src/main/java/org/apache/hadoop/hbase/ipc/MetricsHBaseServer.java [hunk 1]
```diff
@@ -129,8 +130,13 @@ public void exception(Throwable throwable) {
...
         source.callDroppedException();
-      } else if (LOG.isDebugEnabled()) {
-        LOG.debug("Unknown exception type", throwable);
+      } else if (throwable instanceof RequestTooBigException) {
+        source.requestTooBigException();
+      } else {
+        source.otherExceptions();
+        if (LOG.isDebugEnabled()) {
+          LOG.debug("Unknown exception type", throwable);
+        }
```
Comment (pankaj72981) [inline hbase-server/src/main/java/org/apache/hadoop/hbase/ipc/MetricsHBaseServer.java:139]: Need to handle this in ThriftMetrics#exception() also.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m  8s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 24s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   2m  5s |  master passed  |
| +1 :green_heart: |  compile  |   3m  0s |  master passed  |
| +1 :green_heart: |  checkstyle  |   1m  4s |  master passed  |
| +1 :green_heart: |  spotbugs  |   2m 18s |  master passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 10s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   2m  3s |  the patch passed  |
| +1 :green_heart: |  compile  |   3m  3s |  the patch passed  |
| +1 :green_heart: |  javac  |   3m  3s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   0m  6s |  The patch passed checkstyle in hbase-hadoop-compat  |
| +1 :green_heart: |  checkstyle  |   0m 33s |  hbase-server: The patch generated 0 new + 12 unchanged - 1 fixed = 12 total (was 13)  |
| +1 :green_heart: |  checkstyle  |   0m 24s |  The patch passed checkstyle in hbase-thrift  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  hadoopcheck  |  11m 43s |  Patch does not cause any errors with Hadoop 3.1.2 3.2.2 3.3.1.  |
| +1 :green_heart: |  spotbugs  |   2m 37s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 20s |  The patch does not generate ASF License warnings.  |
|  |   |  36m 19s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4248/2/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/4248 |
| Optional Tests | dupname asflicense javac spotbugs hadoopcheck hbaseanti checkstyle compile |
| uname | Linux fd020516d3f3 5.4.0-1025-aws #25~18.04.1-Ubuntu SMP Fri Sep 11 12:03:04 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / ab4351a155 |
| Default Java | AdoptOpenJDK-1.8.0_282-b08 |
| Max. process+thread count | 60 (vs. ulimit of 30000) |
| modules | C: hbase-hadoop-compat hbase-server hbase-thrift U: . |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4248/2/console |
| versions | git=2.17.1 maven=3.6.3 spotbugs=4.2.2 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by sunhelly** [comment]
Hi, @pankaj72981, pr updated, please take a look at your convenience, thanks.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m  9s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  2s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 10s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   2m 41s |  master passed  |
| +1 :green_heart: |  compile  |   1m 27s |  master passed  |
| +1 :green_heart: |  shadedjars  |   3m 40s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   1m 13s |  master passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 13s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   2m 33s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m 23s |  the patch passed  |
| +1 :green_heart: |  javac  |   1m 23s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   3m 36s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   1m 11s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   0m 38s |  hbase-hadoop-compat in the patch passed.  |
| +1 :green_heart: |  unit  | 177m 25s |  hbase-server in the patch passed.  |
| +1 :green_heart: |  unit  |   8m 36s |  hbase-thrift in the patch passed.  |
|  |   | 207m 47s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4248/2/artifact/yetus-jdk11-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/4248 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux 7c145d56ae71 5.4.0-90-generic #101-Ubuntu SMP Fri Oct 15 20:00:55 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / ab4351a155 |
| Default Java | AdoptOpenJDK-11.0.10+9 |
|  Test Results | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4248/2/testReport/ |
| Max. process+thread count | 2512 (vs. ulimit of 30000) |
| modules | C: hbase-hadoop-compat hbase-server hbase-thrift U: . |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4248/2/console |
| versions | git=2.17.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Review by pankaj72981 — APPROVED** [review]
LGTM

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 10s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 14s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   3m 27s |  master passed  |
| +1 :green_heart: |  compile  |   1m 50s |  master passed  |
| +1 :green_heart: |  shadedjars  |   5m 26s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   1m 23s |  master passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 16s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   3m 22s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m 53s |  the patch passed  |
| +1 :green_heart: |  javac  |   1m 53s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   5m 25s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   1m 21s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   0m 54s |  hbase-hadoop-compat in the patch passed.  |
| +1 :green_heart: |  unit  | 242m  5s |  hbase-server in the patch passed.  |
| +1 :green_heart: |  unit  |   8m 34s |  hbase-thrift in the patch passed.  |
|  |   | 279m  2s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4248/2/artifact/yetus-jdk8-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/4248 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux 476daff1e893 5.4.0-90-generic #101-Ubuntu SMP Fri Oct 15 20:00:55 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / ab4351a155 |
| Default Java | AdoptOpenJDK-1.8.0_282-b08 |
|  Test Results | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4248/2/testReport/ |
| Max. process+thread count | 2380 (vs. ulimit of 30000) |
| modules | C: hbase-hadoop-compat hbase-server hbase-thrift U: . |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4248/2/console |
| versions | git=2.17.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Pull request merged**